language/move-stdlib/sources/global.move

```diff
@@ -0,0 +1,35 @@
+/// Represent references to global memory which can freely passed around functions. In core Move, references
+/// like `&R` and `&mut R` can only be passed downwards the stack if obtained from global memory. The types here
+/// relax this restriction, for the cost deferring some checks to runtime instead of compile time.
+module std::global {
+
+    /// Represents an immutable reference to a global. This value can be copied and dropped but not stored.
+    struct Ref<phantom T> has drop, copy {
+        handle: u64
+    }
+
+    /// Represents a mutable reference to global. This value is linear and cannot be stored.
+    struct RefMut<phantom T> has drop {
+        handle: u64
+    }
+
+    /// Borrows the resource at `addr` and returns a `Ref<T>` for it.
+    /// - This function will not type check if there is an active `&T` (indicated by the informal acquires)
+    /// - This function aborts if there is any active `RefMut<T>` for this location.
+    public native fun borrow<T:key>(addr: address): Ref<T> /* acquires T */;
```

Comment:
Is this method breaking the `borrow_global<T>` to the `T` limit? Such as, `T` must be defined in the caller's current module.